packages/next-swc/crates/next-core/src/next_server/resolve.rs

```diff
@@ -18,19 +18,29 @@ use turbo_binding::{
 };
 use turbo_tasks::primitives::{BoolVc, StringsVc};
 
+/// Mark modules as external, so they're resolved at runtime instead of bundled.
+///
+/// If `external_by_default` is set (true) all modules not listed in the
+/// `exceptions` are marked as external as long as it's possible to resolve them
+/// at runtime.
+///
+/// If `external_by_default` is not set (false) only modules listed in the
+/// `exceptions` are marked as external (with the same constraint).
 #[turbo_tasks::value]
 pub(crate) struct ExternalCjsModulesResolvePlugin {
     root: FileSystemPathVc,
-    transpiled_packages: StringsVc,
+    external_by_default: bool,
+    exceptions: StringsVc,
```

Comment:
I find it counter-intuitive that `exceptions` has two meanings depending on another flag `external_by_default`.

This could be an enum instead:

```
enum ExternalPredicate {
    AllExcept(GlobVc),
    Only(GlobVc),
}
```

Unfortunately we don't have a clean way to express this in turbo tasks for task inputs...